File: src/types/clipboard.ts
/**
 * Clipboard Types
 * Type definitions for clipboard history management
 */

/**
 * Clipboard content types
 */
export type ClipboardContentType = 'Text' | 'Image' | 'Html' | 'File';

/**
 * Clipboard item - represents a single clipboard history entry
 */
export interface ClipboardItem {
  id: string;                    // Unique identifier (UUID)
  content_type: ClipboardContentType;  // Type of clipboard content
  text?: string;                 // Text content (for Text/Html types)
  image_path?: string;           // Path to stored image file
  hash: string;                  // Content hash for deduplication (SHA-256)
  timestamp: number;             // When the content was copied (Unix timestamp)
  is_sensitive: boolean;         // Whether content is sensitive (passwords, API keys)
  app_source?: string;           // Application that copied the content
}

/**
 * Clipboard settings
 */
export interface ClipboardSettings {
  max_items: number;                    // Maximum clipboard history items (default: 1000)
  retention_days: number;               // Days to keep non-sensitive items (default: 30)
  sensitive_retention_minutes: number;  // Minutes to keep sensitive items (default: 2)
  enable_image_support: boolean;        // Whether to support image clipboard items
}

File: src-tauri/src/models/clipboard.rs
/**
 * Clipboard Item Model
 * Represents a clipboard history item
 */

use serde::{Deserialize, Serialize};
use std::path::PathBuf;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ClipboardItem {
    pub id: String,
    pub content_type: ClipboardContentType,
    pub text: Option<String>,
    pub image_path: Option<PathBuf>,
    pub hash: String,
    pub timestamp: i64,
    pub is_sensitive: bool,
    pub app_source: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ClipboardContentType {
    Text,
    Image,
    Html,
    File,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ClipboardSettings {
    pub max_items: usize,
    pub retention_days: i64,
    pub sensitive_expiry_minutes: i64,
    pub enabled: bool,
}

impl Default for ClipboardSettings {
    fn default() -> Self {
        Self {
            max_items: 1000,
            retention_days: 30,
            sensitive_expiry_minutes: 2,
            enabled: true,
        }
    }
}
